api/storage_service.cc

```diff
@@ -1428,8 +1428,10 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
                     return cm.perform_sstable_scrub(cf.as_table_state(), opts);
                 }, std::make_optional(sstables::compaction_counters{}), reduce_compaction_counters);
             }, std::make_optional(sstables::compaction_counters{}), reduce_compaction_counters);
-        }).then([] (auto f) {
-            if (f.value().validation_errors == 0) {
+        }).then_wrapped([] (auto f) {
+            if (f.failed()) {
+                return make_ready_future<json::json_return_type>(static_cast<int>(scrub_status::aborted));
+            } else if (f.get().value().validation_errors == 0) {
```

Comment:
`f.get()->validation_errors` is simpler.

Reply:
and I suggest to simplify the if/then ladder by swapping the conditional terms, like this:
```c++
        } else if (f.get()->validation_errors) {
            return make_ready_future<json::json_return_type>(static_cast<int>(scrub_status::validation_errors));
        } else {
                return make_ready_future<json::json_return_type>(static_cast<int>(scrub_status::successful));
        }
````

This makes the relationship between the validation_errors in the result and the validation_errors status clear and direct.
(and it looks like it needs to be done in the earlier patch that introduced this order)